### ISSUE TITLE
Feature disable animation on certain devices

### ISSUE THREAD

**Issue by buepro** [Twikito/onscroll-effect#20]
Animations might not be effective on certain devices. Usually on desktop they make sense where on mobile devices they might not allways be so useful. To adapt to this use case an other data attribute could be introduced. In the library [AOS a `disable` property](https://github.com/michalsnik/aos#1-initialize-aos) is used for that case. Applied to this library a possible attribute could be `data-scroll-disable="mobile"`. The value assigned to the attribute would indicate the biggest device type the animation is disabled for: Setting it to `tablet` would disable it for mobiles and tablets. I found that AOS is using for this a [`Detector` class](https://github.com/michalsnik/aos/blob/next/src/js/helpers/detector.js).

**Comment by Twikito** [comment]
Hi @buepro 👋 
Thanks for taking time adding this issue.

Since this library is about adding one or more css classes, don't you think managing if an animation should be launched or not on certain device should be done with css media queries?

It's y opinion, but what do you think?

**Comment by buepro** [comment]
Hi @Twikito

Good point. In this context the question arises if beside animations there are other use cases where it is desired to assign classes depending on the device. 

At least for animations such a feature seems to be of interest since [AOS](https://github.com/michalsnik/aos#1-initialize-aos) as well as [Josh.js](https://github.com/mamunhpath/josh.js#advanced-usage) provide it.

The advantages from such a feature would be 
- that more extensive media queries (covering all kind of devices in landscape and portrait) wouldn't be needed
- and the behaviour is controlled at one point

**Comment by Twikito** [comment]
Hi @buepro 

Ok, I see.

I'm thinking about using this https://www.npmjs.com/package/device. What do you think?

**Comment by buepro** [comment]
Hi @Twikito 

This seems to be good and actively maintained. 

What do you think about just checking for a minimum screen square? For example with

`let enableAnimation = window.matchMedia("only screen and (min-width: minScreenLength) and (min-height: minScreenLength)").matches;`

where `minScreenLength` comes from the data attribute (e.g. `data-scroll-screen="900"`).

**Comment by Twikito** [comment]
Hey @buepro 👋 

Well, since device detection is only possible with javascript, it make sence to add this feature, but dealing with screen size should only be managed in css in my opinion. Or maybe you have an example that shows me I'm wrong?

**Comment by buepro** [comment]
You're right, to use the device package seems to be the best choice here.

**Comment by Twikito** [comment]
Hey @buepro 👋 

I've made some tests, but for now, everything I've tried increased the weight of the script by two or even three. In my opinion, it's unacceptable for only one feature.

So I've search a script that could do the job, but in a lightly way. I mean, I only need the device type, not everything else.

I finally found this script: https://dev.to/itsabdessalam/detect-current-device-type-with-javascript-490j
After several tests, it looks like working just the way I need.

What do you think? Could you test this script by your side?

**Comment by buepro** [comment]
Hi @Twikito 

That looks good. In AOS and Josh.js as well `navigator.userAgent` is evaluated for that purpose. In AOS a function is used to get the user agent string (`var ua = navigator.userAgent || navigator.vendor || window.opera || '';`) and the regular expression string is more extensive (see [detector.js](https://github.com/michalsnik/aos/blob/next/src/js/helpers/detector.js)).

Tests on my side would cover too view devices hence wouldn't contribute to a reliable solution. For this I would rather suggest to use something that is already used in the field. It seems that the AOS detection is inspired by [hgoebl/mobile-detect.js/../mobile-detect.js](https://github.com/hgoebl/mobile-detect.js/blob/0a9ae49fa60c311f785d0fff0526c3e380a37c12/mobile-detect.js#L298).

The post from [Abdessalam](https://dev.to/itsabdessalam/detect-current-device-type-with-javascript-490j) is good but to be able to maintain the feature it might be more helpful to implement something referencing an active package (like `hgoebl/mobile-detect.js`).
